### PR TITLE
Add validation event check force delete to a job

### PR DESCRIPTION
**Pull request by thiagoyoussef**

### CODE REVIEW

File: app/jobs/validation_event_check_force_delete_job.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class ValidationEventCheckForceDeleteJob < ApplicationJob
```
Comment (thiagoyoussef) [inline app/jobs/validation_event_check_force_delete_job.rb:1]: change the name

File: lib/tasks/check_force_delete.rake [hunk 2]
```diff
@@ -0,0 +1,11 @@
+desc 'Check Force Delete'
+task check_force_delete: :environment do
+  invalid_contacts = Contact.joins(:validation_events).select do |contact|
+    events = contact.validation_events
+    events.mx.count >= 3 || events.regex.present?
```
Comment (thiagoyoussef) [inline lib/tasks/check_force_delete.rake:5]: Use `MX_CHECK`

Comment (thiagoyoussef) [reply lib/tasks/check_force_delete.rake:5]: invalid

File: lib/tasks/check_force_delete.rake [hunk 3]
```diff
@@ -0,0 +1,11 @@
+desc 'Check Force Delete'
+task check_force_delete: :environment do
```
Comment (thiagoyoussef) [inline lib/tasks/check_force_delete.rake:2]: Add args to use batches and schedule delay between jobs

**Review by OlegPhenomenon — APPROVED** [review]

**Comment by vohmar** [comment]
To run the job you need to execute the rake task check_force_delete
3 optional arguments to call the task:

- `batch` - value can be true or false - default value is false - true: executes the job in batches, one job = many contacts - false: one job = one contact - this way Martin can see which one performs better
- `batch_size` - number of contacts that will be performed in each job - default 1000
- `batches_delay` - minutes between each job that will be performed - default 15 minutes

It's possible to call the task like this:

-  `rake check_force_delete` - will perform without batch, one contact = one job
-  `rake 'check_force_delete[true, 10000]'` - will perform with batch, 10000 contacts = one job, with 15 minutes of difference between jobs
-  `rake 'check_force_delete[true, 10000, 30]'` - will perform with batch, 10000 contacts = one job, with 30 minutes of difference between jobs

**Pull request merged**